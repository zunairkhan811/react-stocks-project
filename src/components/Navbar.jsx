import { NavLink } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faArrowLeft, faMicrophone } from '@fortawesome/free-solid-svg-icons';

const Navbar = () => (
  <nav>
    <NavLink to="/"><FontAwesomeIcon icon={faArrowLeft} data-testid="mock-test1" /></NavLink>
    <NavLink to="/">Stocks</NavLink>
    <NavLink to="/"><FontAwesomeIcon icon={faMicrophone} data-testid="mock-test2"  /></NavLink>
  </nav>
);

export default Navbar;
