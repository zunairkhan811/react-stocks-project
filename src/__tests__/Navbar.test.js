import renderer from "react-test-renderer";
import Navbar from "../components/Navbar";
import { BrowserRouter } from "react-router-dom";
import '@testing-library/jest-dom';

it('Navbar renders correctly',()=>{
    const tree = renderer.create(
        
        <BrowserRouter>
          <Navbar />  
        </BrowserRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
})