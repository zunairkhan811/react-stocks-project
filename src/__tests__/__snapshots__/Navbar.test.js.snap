// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
<nav>
  <a
    aria-current="page"
    className="active"
    href="/"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-arrow-left "
      data-icon="arrow-left"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.2 288 416 288c17.7 0 32-14.3 32-32s-14.3-32-32-32l-306.7 0L214.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </a>
  <a
    aria-current="page"
    className="active"
    href="/"
    onClick={[Function]}
  >
    Stocks
  </a>
  <a
    aria-current="page"
    className="active"
    href="/"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-microphone "
      data-icon="microphone"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 384 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M192 0C139 0 96 43 96 96V256c0 53 43 96 96 96s96-43 96-96V96c0-53-43-96-96-96zM64 216c0-13.3-10.7-24-24-24s-24 10.7-24 24v40c0 89.1 66.2 162.7 152 174.4V464H120c-13.3 0-24 10.7-24 24s10.7 24 24 24h72 72c13.3 0 24-10.7 24-24s-10.7-24-24-24H216V430.4c85.8-11.7 152-85.3 152-174.4V216c0-13.3-10.7-24-24-24s-24 10.7-24 24v40c0 70.7-57.3 128-128 128s-128-57.3-128-128V216z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </a>
</nav>
`;
